### PR TITLE
Patch reviews-api

### PR DESCRIPTION
**Pull request by adnaani**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -83,23 +102,56 @@ describe("API: /api/reviews", () => {
...
+      const review_id = 1;
+      const inc_vote = { inc_votes: "NaN" };
+
       return request(app)
-        .get(`/api/reviews/${review_id}`)
+        .patch(`/api/reviews/${review_id}`)
         .expect(400)
+        .send(inc_vote)
         .then(({ body: { message } }) => {
           expect(message).toBe("id is not valid");
```
Comment (popatre) [inline __tests__/app.test.js]: This message, doesnt match whats caused the error. They havent passed an id, they've passed not a number for the inc votes

File: __tests__/app.test.js [hunk 2]
```diff
@@ -83,23 +102,56 @@ describe("API: /api/reviews", () => {
...
+
+    test.only("404: responds with an error message when passed an endpoint with correct data type but does not exist", () => {
       const review_id = 999;
+      const inc_vote = { inc_votes: "3" };
+
       return request(app)
-        .get(`/api/reviews/${review_id}`)
+        .patch(`/api/reviews/${review_id}`)
         .expect(404)
+        .send(inc_vote)
```
Comment (popatre) [inline __tests__/app.test.js:166]: - [ ] TEST - 200/400 inc_vote key missing - either respond with an error or make no change.

**Review by popatre — CHANGES_REQUESTED** [review]
Just some changes and a test to add

File: __tests__/app.test.js (hunk 2 above)
Comment (adnaani) [reply __tests__/app.test.js:166]: Thanks Jonathan I will make these changes now!!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -89,4 +87,87 @@ describe("API: /api/reviews", () => {
...
+  describe("PATCH: /api/reviews/:review_id", () => {
+    test("201: responds with a votes incrementing by inc_vote", () => {
+      const review_id = 1;
+
+      return request(app)
+        .patch(`/api/reviews/${review_id}`)
+        .send({ inc_votes: 5 })
+        .expect(201)
+        .then(({ body: { review } }) => {
+          expect(review).toBeInstanceOf(Object);
```
Comment (mallionaire) [inline __tests__/app.test.js]: Personally I don't think you gain much by checking the data type here, you can infer this from the fact that you're checking keys with your other expect statements. 
Its not bad or wrong, I'd just have a think about what confidence you're gaining 😄

File: model/reviews.js [hunk 4]
```diff
@@ -13,3 +13,23 @@ exports.selectReviewById = (review_id) => {
       return rows[0];
     });
 };
+
+exports.updateReviewById = (review_id, { inc_votes } = 0) => {
+  return db
+    .query(
+      `UPDATE reviews
```
Comment (mallionaire) [inline model/reviews.js:24]: This is super clear to read, lovely

File: model/reviews.js [hunk 5]
```diff
@@ -13,3 +13,23 @@ exports.selectReviewById = (review_id) => {
...
+    SET votes = votes + $1
+    WHERE review_id = $2
+    RETURNING*`,
```
Comment (mallionaire) [inline model/reviews.js]: Am I gonna kick off about missing whitespace... I might, I'd change it. You can decide whether it bothers you haha

**Review by mallionaire — APPROVED** [review]
Really nice style here, I particularly like that you're very consistent with destructuring from the response, and passing values to your endpoints in your tests. 
Left you some thoughts more than feedback/changes - you can decide then go ahead and merge 😄

File: model/reviews.js (hunk 5 above)
Comment (adnaani) [reply model/reviews.js]: Thanks for the amazing feedback!! I will take note of everything and try implement it through out my code. Also I am a bit picky so I will have to use the whitespace ahah. Thanks, again Haz!!

**Review by hevdevs — APPROVED** [review]
nice refactoring of error handling funcs, all looking good!

**Pull request merged**